main.py

```diff
@@ -138,41 +139,37 @@ def get_recent_projects(self, ide_key: IdeKey) -> list[IdeProject]:
         if base_path is None or not os.path.isdir(base_path):
             raise FileNotFoundError("Cant find configs directory")
 
-        versions: list[semver.VersionInfo] = []
+        versions: Dict[str, semver.VersionInfo] = {}
         for path in os.listdir(base_path):
-            match = re.match(
-                f"^{ide_data.config_prefix}" +
-                r"(?P<major>0|[1-9]\d*)(\.(?P<minor>0|[1-9]\d*)(\.(?P<patch>0|[1-9]\d*))?)?",
-                path)
+            if os.path.exists(os.path.join(base_path, path, "options", ide_data.recent_projects_file)):
+                for config_prefix in ide_data.config_prefixes:
+                    match = re.match(
+                        f"^{config_prefix}" +
+                        r"(?P<major>0|[1-9]\d*)(\.(?P<minor>0|[1-9]\d*)(\.(?P<patch>0|[1-9]\d*))?)?",
+                        path)
 
-            if match is not None:
-                version_dict = {
-                    key: 0 if value is None else value for key, value in match.groupdict().items()
-                }
+                    if match is not None:
+                        version_dict = {
+                            key: 0 if value is None else value for key, value in match.groupdict().items()
+                        }
 
-                versions.append(semver.VersionInfo(**version_dict))
+                        versions[path] = semver.VersionInfo(**version_dict)
 
         if len(versions) == 0:
             return []
 
-        version = max(versions)
+        version = max(versions, key=versions.get)
         config_dir = os.path.join(
             base_path,
-            f"{ide_data.config_prefix}{version.major}.{version.minor}",
+            version,
```

Comment:
Correct me if I'm wrong but doesn't this evaluate to just the version without the config prefix? For example, instead of returning `CLion2023.2` it will now return `2023.2`

Reply:
Since versions is now a dict instead of a list max() returns the key with the maximum value. This key is the name of the directory.

I can rename the variable if you want since it no longer actually contains the version, but the directory name for the highest version instead.

Reply:
Oh, you are right, I remember now... the `max()` function takes an iterable which in the case of a dict would be the `key`, but since you specified to index it by `versions.get()` method it will compare the values using that method, but still return the `key` of the max value.
Anyway, if that is the case then a rename of the variable would avoid any confusion.

Reply:
Done